tests/rptest/cluster/k8s.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2014 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+
+from ducktape.cluster.cluster_spec import ClusterSpec, WINDOWS
+from ducktape.cluster.node_container import NodeContainer
+from ducktape.command_line.defaults import ConsoleDefaults
+from ducktape.cluster.cluster import Cluster, ClusterNode
+from ducktape.cluster.linux_remoteaccount import LinuxRemoteAccount
+from ducktape.cluster.windows_remoteaccount import WindowsRemoteAccount
+from ducktape.cluster.remoteaccount import RemoteAccountSSHConfig
+
+import json
+import os
+import traceback
+
+
+class K8sJsonCluster(Cluster):
```

Comment:
should this go into ducktape upstream?

Reply:
since this is going to be a JSON file that has data for specifically connecting to redpanda k8s clusters and the data will be used for specific `tests/rptest/k8s_tests` tests, i felt it would be more sensible to keep it in `redpanda` repo. it will also make future rebase of `ducktape` with upstream easier and save on the whole dev cycle of updating ducktape for and changing git commit in 2 places to realize any `K8sJsonCluster` changes.

but willing to move this file into the ducktape fork repo later if it starts to evolve into something that makes more sense to put there.